templates/openstackclient/bin/tripleo-deploy.sh

```diff
@@ -94,6 +95,41 @@ play() {
 
   cd tripleo-ansible
 
+  STACK_ACTION_ARG=""
+  if [ "${OSP_VERSION}" = "16.2" ]; then
+    # We need stack_action to be set correctly in 16.2 as the network_config and octavia roles in tripleo-ansible
+    # rely on it.
+    # As we can't use the existance of a heat stack, instead use the existance of /var/lib/tripleo-config on any
+    # overcloud host (created by the tripleo-bootstrap role) to determine if this is a "CREATE" or an "UPDATE"
+    cat <<EOF > set_stack_action_playbook.yaml
+{{` ---
+- hosts: overcloud
+  name: Determine the correct stack action for 16.2 deployments
+  gather_facts: false
+  tasks:
+    - stat:
+        path: /var/lib/tripleo-config
+      register: deploy_exists
+      ignore_unreachable: yes
+    - set_fact:
+        deploy_exists: "{{ hostvars['localhost'].deploy_exists|default('0')|int + 1 }}"
+      when: deploy_exists.stat.exists|default(False)
+      delegate_to: localhost
+      delegate_facts: yes
+    - copy:
+        dest: stack_action_override.yaml
+        content: |
+          stack_action: {{ 'UPDATE' if hostvars['localhost'].deploy_exists|default('0')|int > 0 else 'CREATE' }}
+      delegate_to: localhost
+      run_once: yes
+EOF`}}
+
+    ansible-playbook \
```

Comment:
TODO: the initial deploy could be split (e.g --limit Controller for the 1st deploy then --limit Compute for the 2nd) so safer to run with the same --limit args as the deploy we are about to run